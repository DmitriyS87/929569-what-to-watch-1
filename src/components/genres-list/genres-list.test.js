import React from 'react';
import renderer from 'react-test-renderer';
import GenresList from '../genres-list/genres-list';
import TEST_MOCKS from '../../mocks/test-mocks.js';

it(`GenresList correctly renders after relunch`, () => {
  const mockHandle = jest.fn();
  const tree = renderer
    .create(<GenresList
      movies={TEST_MOCKS.films}
      activeItem={TEST_MOCKS.activeGenre}
      setGenre={mockHandle}
      setActive={mockHandle}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
