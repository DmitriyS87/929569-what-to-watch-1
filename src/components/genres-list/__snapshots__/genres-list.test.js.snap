// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresList correctly renders after relunch 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Adventure
    </a>
  </li>
</ul>
`;
